### DISCUSSION TITLE
Default behavior of the Alpha Blend mode

### DISCUSSION THREAD

**Discussion by vinceFix** [ggarra13/mrv2#139] [Q&A] [answered]
Hi community,
Do you have a way to force the Alpha Blend mode to be 'None' by default ? I couldn't find any option in the preferences, nor in the command line arguments.

Thanks !

**Comment by BigRoy** [comment]
#### This does not provide an answer but it gives more context to your question about what might have been possible ways to do so

I don't know of any preference/settings for this unfortunately. However you _could_ set it through the Python API:
```python
from mrv2 import cmd, image

options = cmd.imageOptions()
options.alphaBlend = getattr(image.AlphaBlend, "None")  # see issue: https://github.com/ggarra13/mrv2/issues/140
cmd.setImageOptions(options)
```

The command line does allow running a `-pythonScript` but it will run and exit after the code finishes - so isn't a viable solution for this case. 

You can however define a custom Python plug-in that you make available to MR2, in that file you can **also** run your own python code as MRV2 initializes. **However** I tried this code snippet above in a custom python plug-in and it'll make MRV2 fail to launch:

```
GOT SIGNAL 11
0: RtAudio::getDefaultInputDevice - 0x140696318165296
1: seh_filter_exe - 0x140720941501632
2: RtApi::verifyStream - 0x140696321790800
3: _C_specific_handler - 0x140720850725936
4: _chkstk - 0x140720982926016
5: RtlRaiseException - 0x140720982593584
6: KiUserExceptionDispatcher - 0x140720982920896
7: RtApi::closeStream - 0x140696316111120
```

So unfortunately that also doesn't seem like a viable option either.

For more details on the **Plug-in System** page in the Documentation (Help > Documentation menu entry in MRV2) or see [this issue](https://github.com/ggarra13/mrv2/issues/68) for more details.

#### Feature request

It might be worth for the Plugin system to expose a method that gets triggered after the UI initializes to allow custom setting customizations, etc. E.g.

```python
from mrv2 import cmd, image, plugin

class MyPlugin(plugin.Plugin):
    def on_ui_initialized(self):
        options = cmd.imageOptions()
        options.alphaBlend = getattr(image.AlphaBlend, "None")
        cmd.setImageOptions(options)
```
or maybe called `post_initialize()` or alike.

**Comment by ggarra13** [comment, marked as answer]
Ok.  I don't have a settings in the preferences or a python solution yet but I have a work-around for now.   First, download the latest beta (v0.8.2) from sourceforge:

https://sourceforge.net/projects/mrv2/files/beta/

Open mrv2 without any images loaded.  Go to the menus and set the alpha blending to what you want (None, right?).  Save the session (it is an .mrv2s file in JSON format).

Now, whenever you want to start with alpha blending in None, you open the session file (from the UI) or from the command-line like:

mrv2 test.mrv2s test.mov test.0001.exr

..etc..

**Reply by vinceFix** [reply]
Haha, that'a a nice trick @ggarra13 !
Thank you !

**Comment by ggarra13** [comment]
If you want, you can download the latest beta of v0.8.2 from the sourceforge site.  It adds a preference section to control video levels and alpha blend in Preferences (F10)->User Interface->View Window.

**Reply by vinceFix** [reply]
I'm a bit late but thank you so much @ggarra13, it's really useful !
